packages/expo-firebase-analytics/android/src/main/java/expo/modules/firebase/analytics/FirebaseAnalyticsModule.java

```diff
@@ -0,0 +1,193 @@
+        }
+        
+        return constants;
+    }
+    
+    @ExpoMethod
+    public void initializeAppDangerously(final Map<String, String> options, Promise promise) {
+        try {
+            FirebaseOptions.Builder builder = new FirebaseOptions.Builder();
+
+            builder.setApiKey(options.get("apiKey"));
+            builder.setApplicationId(options.get("appId"));
+            builder.setProjectId(options.get("projectId"));
+            builder.setDatabaseUrl(options.get("databaseURL"));
+            builder.setStorageBucket(options.get("storageBucket"));
+            builder.setGcmSenderId(options.get("messagingSenderId"));
+            builder.setGaTrackingId(options.get("trackingId"));
+
+            FirebaseApp firebaseApp = getDefaultApp();
+            if (firebaseApp != null) {
+                FirebaseOptions currentOptions = firebaseApp.getOptions();
+                if (!currentOptions.getApiKey().equals(options.get("apiKey")) ||
+                        !currentOptions.getApplicationId().equals(options.get("appId"))) {
+                    firebaseApp.delete();
+                } else {
+                    promise.resolve(null);
+                    return;
+                }
+            }
+            FirebaseApp.initializeApp(mActivity.getApplicationContext(), builder.build());
+            promise.resolve(null);
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    private @Nullable
+    FirebaseApp getDefaultApp() {
+        try {
+            return FirebaseApp.getInstance();
+        } catch (Exception ignored) {
+        }
+        return null;
+    }
+
+    @ExpoMethod
+    public void deleteApp(final Map<String, String> options, Promise promise) {
+        try {
+            FirebaseApp firebaseApp = getDefaultApp();
+            if (firebaseApp != null) {
+                firebaseApp.delete();
+            }
+            promise.resolve(null);
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    @ExpoMethod
+    public void logEvent(final String name, @Nullable Map<String, Object> params, Promise promise) {
+        try {
+            FirebaseAnalytics.getInstance(mActivity.getApplicationContext()).logEvent(name, new MapArguments(params).toBundle());
+            promise.resolve(null);
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    @ExpoMethod
+    public void setAnalyticsCollectionEnabled(final Boolean enabled, Promise promise) {
+        try {
+            FirebaseAnalytics.getInstance(mActivity.getApplicationContext()).setAnalyticsCollectionEnabled(enabled);
+            promise.resolve(null);
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    @ExpoMethod
+    public void setCurrentScreen(final String screenName, final String screenClassOverride, final Promise promise) {
+        mActivity.runOnUiThread(new Runnable() {
```

Comment:
If this is the only method that is required to be run on UI Thread, would you care to add a comment here that not running other methods on UI thread is expected?